### PR TITLE
docs: refactor and update security

### PR DESCRIPTION
**Pull request by james-d-elliott**
Refactors the secrurity documentation to be up-to-date and conform to our style guidelines. Additionally went over each part and reworded things that needed it.

### CODE REVIEW

File: docs/security/measures.md [hunk 1]
```diff
@@ -19,132 +19,147 @@ Read about these attributes in detail on the
...
-the login. Lastly the absolute minimum time authentication can take is 250ms. Both of these measures
-also have the added effect of creating an additional delay for all authentication attempts reducing
-the likelihood a password can be brute-forced even if regulation settings are too permissive.
+When Authelia first starts it assumes the last 10 attempts took 1000 milliseconds each. As users login successfully it
+quickly adjusts to the actual time the login attempts take. This process is independent of the login backend you have
+configured.
+
+The cost of this is low since in the instance of a user not existing it just stops processing the request to delay the
+login. Lastly the absolute minimum time authentication can take is 250 milliseconds. Both of these measures also have 
+the added effect of creating an additional delay for all authentication attempts reducing the likelihood a password can
```
Comment (clems4ever) [inline docs/security/measures.md:46]: Actually, it does not reduce the likelihood of password brute-force if Authelia can handle an infinite amount of requests in parallel so maybe we should simply remove this sentence. (That's why I've just created the ticket around DDoS :) )

File: docs/security/measures.md [hunk 2]
```diff
@@ -19,132 +19,147 @@ Read about these attributes in detail on the
...
+information.
 
 ### Session security
 
-We have a few options to configure the security of a session. The main and most important
-one is the session secret. This is used to encrypt the session data when when stored in the
-Redis key value database. This should be as random as possible.
+We have a few options to configure the security of a session. The main and most important one is the session secret.
+This is used to encrypt the session data when it is stored in the [Redis](../configuration/session/redis.md) key value
+database. The value of this option should long and as random as possible. See more in the
```
Comment (clems4ever) [inline docs/security/measures.md:149]:
```suggestion
database. The value of this option should be long and as random as possible. See more in the

File: docs/security/measures.md [hunk 3]
```diff
@@ -254,55 +273,89 @@ bantime = 1d
...
-chown user:group /authelia
-docker run --user user:group -v /authelia:/config authelia/authelia:latest
+
+### Docker user directive
+
+The docker user directive allows you to configure the user the entrypoint runs as. This is generally the most secure
+option for containers as no process accessible to the container ever runs as root which prevents a compromised container
+from exploiting unnecessary privileges.
+
+The directive can applied either be applied in your `docker run` command using the `--user` argument or by
```
Comment (clems4ever) [inline docs/security/measures.md:315]:
```suggestion
The directive can either be applied in your `docker run` command using the `--user` argument or by

File: docs/security/threat-model.md [hunk 4]
```diff
@@ -8,44 +8,76 @@ nav_order: 2
...
 
 ## General assumptions
 
-Authelia is considered to be running within a trusted network and it heavily relies on the first level of security provided by reverse proxies. It's very important that you take time configuring your reverse proxy properly to get all the authentication benefits brought by Authelia.
-Some general security tweaks are listed in [Security Measures](./measures.md) to give you some ideas.
+Authelia is considered to be running within a trusted network and it heavily relies on the first level of security 
+provided by reverse proxies. It's very important that you take time configuring your reverse proxy properly to get all 
+the authentication benefits brought by Authelia.
+Some general security tweaks are listed in [Security Measures](./measures.md#additional-proxy-protection-measures) to 
+give you some ideas.
```
Comment (clems4ever) [inline docs/security/threat-model.md:20]:
```suggestion
and give you some ideas.

**Review by clems4ever — CHANGES_REQUESTED** [review]
Only tiny things but LGTM otherwise. Good job!

File: docs/security/measures.md (hunk 1 above)
Comment (james-d-elliott) [reply docs/security/measures.md:46]: Ah yeah you're right!

File: docs/security/index.md [hunk 5]
```diff
@@ -26,9 +26,16 @@ Join the [Matrix Room](https://riot.im/app/#/room/#authelia:matrix.org) and loca
...
+always just ask for one of the maintainers in the public channels by pinging them. Once you've made contact we ask
+you privately message the maintainer to communicate the vulnerability.
+
 ### Email
 
 You can contact any of the maintainers for security vulnerability related issues by emailing 
 [security@authelia.com](mailto:security@authelia.com). This email is strictly reserved for security and vulnerability
-disclosure related matters. If you need to contact us for another reason please use [Matrix](#matrix) or
-[team@authelia.com](mailto:security@authelia.com).
+disclosure related matters. If you need to contact us for another reason via please use one of the other methods above
```
Comment (nightah) [inline docs/security/index.md:40]:
```suggestion
disclosure related matters. If you need to contact us for another reason please use one of the other methods above
```

File: docs/security/threat-model.md (hunk 4 above)
Comment (nightah) [reply docs/security/threat-model.md:20]: This suggested `and` looks out of place I don't think it's necessary.

**Review by nightah — CHANGES_REQUESTED** [review]
Only one tiny nit I managed to pick up and a recommendation around one of @clems4ever's suggestions.

**Review by nightah — APPROVED** [review]
LGTM.

**Pull request merged**